### ISSUE TITLE
"type 'String' is not a subtype of type 'ResponseBody?' in type cast" when using dio.download.

### ISSUE THREAD

**Issue by Hannnes1** [cfug/dio#1569]
### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          |   flutter   |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |   3.3.1   |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |   4.0.6   |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |   100%   |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

We are using the following code to download a file:

```
dio.download(
        'https://file-examples.com/storage/feb2e515cc6339d7ba1ffcd/2017/10/file-sample_150kB.pdf',
        './savePath/test.pdf',
      );
```

The problem we are having is that we get an error:
```
DioError [DioErrorType.other]: type 'String' is not a subtype of type 'ResponseBody?' in type cast
```

When using an interceptor to log the response, we see that the body is the content of the file in plain text. Are we using `download` wrong, or why do we get this error?

**Comment by sharadgoregore** [comment]
i am facing same issue?? do you know ho to solve??

**Comment by sharadgoregore** [comment]
i am facing same issue?? do you know how to solve??
if i click again then it download but for first time it throws an error

flutter: DioError [DioErrorType.other]: type 'String' is not a subtype of type 'ResponseBody?' in type cast
#0 DioMixin.assureResponse
package:dio/src/dio_mixin.dart:818
https://github.com/cfug/dio/issues/1 DioMixin.fetch.
package:dio/src/dio_mixin.dart:630

https://github.com/cfug/dio/issues/2 DioForNative.download
package:dio/…/entry/dio_for_native.dart:84

**Comment by sharadgoregore** [comment]
@AlexV525  do you know how to solve it ???

**Comment by tijins** [comment]
I got same problem and solved it.

The cause of the error was that option.responseType was not inherited when retrying with Interceptor, and I found that I could avoid the error by explicitly specifying option.responseType.

The code to avoid the error is as follows.
```dart
  Future<Response<dynamic>> _retry(RequestOptions prevOptions) async {
    final options = Options(
        method: prevOptions.method,
        headers: prevOptions.headers,
        receiveDataWhenStatusError: prevOptions.receiveDataWhenStatusError,
        responseType: prevOptions.responseType);
    return _dio.request(prevOptions.path,
        data: prevOptions.data, queryParameters: prevOptions.queryParameters, options: options);
  }

```

**Comment by sharadgoregore** [comment]
Thanks @tijins it worked

**Comment by EdmarS** [comment]
> I got same problem and solved it.
> 
> The cause of the error was that option.responseType was not inherited when retrying with Interceptor, and I found that I could avoid the error by explicitly specifying option.responseType.
> 
> The code to avoid the error is as follows.
> 
> ```dart
>   Future<Response<dynamic>> _retry(RequestOptions prevOptions) async {
>     final options = Options(
>         method: prevOptions.method,
>         headers: prevOptions.headers,
>         receiveDataWhenStatusError: prevOptions.receiveDataWhenStatusError,
>         responseType: prevOptions.responseType);
>     return _dio.request(prevOptions.path,
>         data: prevOptions.data, queryParameters: prevOptions.queryParameters, options: options);
>   }
> ```

I have the same problem, but only on Windows desktop. MacOS and Android/iOS seem to work fine. Where in my code should I apply this code block?
